wagl/metadata.py

```diff
@@ -196,84 +247,57 @@ def load_ancillary(acquisitions, fid, sbt=False):
-    ancillary = load_ancillary(acquisitions, ancillary_group, sbt)
-    for item in ancillary:
-        for remove in ['CLASS', 'VERSION', 'query_date', 'data_source']:
-            ancillary[item].pop(remove, None)
-
-    software_versions = {'wagl': {'version': wagl.__version__,
-                                  'repo_url': 'https://github.com/GeoscienceAustralia/wagl.git'}, # pylint: disable=line-too-long
-                         'modtran': {'version': '6.0.1',
-                                     'repo_url': 'http://www.ontar.com/software/productdetails.aspx?item=modtran'} # pylint: disable=line-too-long
-                        }
-
-    algorithm = {}
-    if sbt:
-        dname = DatasetName.SBT_YAML.value
-        algorithm['sbt_doi'] = 'TODO'
-    else:
-        dname = DatasetName.NBAR_YAML.value
-        algorithm['algorithm_version'] = 2.0
-        algorithm['arg25_doi'] = 'http://dx.doi.org/10.4225/25/5487CC0D4F40B'
-        algorithm['nbar_doi'] = 'http://dx.doi.org/10.1109/JSTARS.2010.2042281'
-        algorithm['nbar_terrain_corrected_doi'] = 'http://dx.doi.org/10.1016/j.rse.2012.06.018' # pylint: disable=line-too-long
+            result.update(load_sbt_ancillary(fid))
+
+        if nbar:
+            for grp_name in res_group_bands:
+                grp_ancillary = load_nbar_ancillary(res_group_bands[grp_name], fid)
+                for item in grp_ancillary:
+                    for remove in ['CLASS', 'VERSION', 'query_date', 'data_source']:
+                        grp_ancillary[item].pop(remove, None)
+                result.update(grp_ancillary)
+
+        return result
+
+    def software_versions():
+        return {'wagl': {'version': wagl.__version__,
+                         'repo_url': 'https://github.com/GeoscienceAustralia/wagl.git'},
+                'modtran': {'version': '6.0.1',
+                            'repo_url': 'http://www.ontar.com/software/productdetails.aspx?item=modtran'}}
+
+    def algorithm():
+        result = {}
+
+        if sbt:
+            result['sbt_doi'] = 'TODO'
+
+        if nbar:
+            result['algorithm_version'] = 2.0
+            result['arg25_doi'] = 'http://dx.doi.org/10.4225/25/5487CC0D4F40B'
+            result['nbar_doi'] = 'http://dx.doi.org/10.1109/JSTARS.2010.2042281'
+            result['nbar_terrain_corrected_doi'] = 'http://dx.doi.org/10.1016/j.rse.2012.06.018'
+
+        return result
 
     metadata = {'system_information': get_system_information(),
-                'source_datasets': source_info,
-                'ancillary': ancillary,
-                'algorithm_information': algorithm,
-                'software_versions': software_versions}
-    
+                'source_datasets': source_info(),
+                'ancillary': ancillary(ancillary_group),
+                'algorithm_information': algorithm(),
+                'software_versions': software_versions(),
+                'id': str(uuid.uuid1()),
```

Comment:
I don't have any major input here.  Maybe @jeremyh will have some thoughts on uuid1.

Reply:
uuid1 looks like it uses the clock + hardware address, since we'll be processing multiple granules on the same hardware address uuid4 appears safer to me.

Reply:
Yes -- use `uuid4()`.

uuid1 is not sufficiently random in our deployments.